web/src/popover_menus.js

```diff
@@ -833,6 +840,15 @@ export function initialize() {
                     format(date.setDate(date.getDate()), "MMM d yyyy ") + send_time;
                 return send_at_time;
             }
+            case "send_later_monday": {
+                const send_time = send_later_monday[send_later_in].time;
+                const date = new Date();
+                // Subtract from 8 to find the next Monday.
+                const monday_offset = 8 - date.getDay();
+                const scheduled_date = date.setDate(date.getDate() + monday_offset);
+                const send_at_time = format(scheduled_date, "MMM d yyyy ") + send_time;
+                return send_at_time;
+            }
             // No default
         }
```

Comment:
I think it's probably worth extracting this switch statement, probably including the `send_later_monday` type functions, as a function in `scheduled_messages.js`, and writing some node tests for it -- this kind of fiddly date logic is the kind of thing that can be wrong or break, and it seems like it has the kind of interface that is quite suitable for testing.

(If that creates import cycles, I think we shouldn't worry about it -- those can be resolved once the popover_menus library supports popovers being declared in another module)

Reply:
That makes sense. I'll give this a shot.